tokenprovider/token_provider_gce.go

```diff
@@ -3,18 +3,33 @@ package tokenprovider
 import (
 	"context"
 
+	"golang.org/x/oauth2"
 	"golang.org/x/oauth2/google"
 )
 
+type gceSource struct {
+	cacheKey string
+	scopes   []string
+}
+
 // NewGceAccessTokenProvider returns a token provider for gce authentication
-func NewGceAccessTokenProvider(ctx context.Context, cfg *Config) (TokenProvider, error) {
-	source, err := google.DefaultTokenSource(ctx, cfg.Scopes...)
+func NewGceAccessTokenProvider(cfg Config) TokenProvider {
+	return &tokenProviderImpl{
+		&gceSource{
+			cacheKey: createCacheKey("gce", &cfg),
+			scopes:   cfg.Scopes,
+		},
+	}
+}
+
+func (source *gceSource) getCacheKey() string {
+	return source.cacheKey
+}
```

Comment:
I guess it's a matter of style but if both the function and the property are not exported, what's the point of defining the method?